issuer/src/main.rs

```diff
@@ -745,67 +758,135 @@ fn get_derivation_origin(_hostname: &str) -> Result<DerivationOriginData, Deriva
     }
 }
 
-fn verify_spec_and_get_group_name(spec: &CredentialSpec) -> Result<(String, Principal), String> {
-    if spec.credential_type.as_str() == "VerifiedMember" {
-        let Some(arguments) = &spec.arguments else {
-            return Err("Credential spec has no arguments".to_string());
-        };
-        if arguments.len() != 2 {
-            return Err("Credential spec has unexpected arguments".to_string());
-        }
-        let group_name_arg = "groupName";
-        let Some(value) = arguments.get(group_name_arg) else {
-            return Err(format!(
-                "Credential spec has no {}-argument",
-                group_name_arg
-            ));
-        };
-        let ArgumentValue::String(group_name) = value else {
-            return Err(format!(
-                "Credential spec has an unexpected value for {}-argument",
-                group_name_arg
-            ));
-        };
+fn get_int_arg_value(arg_name: &str, spec: &CredentialSpec) -> Result<i32, String> {
+    let Some(arguments) = &spec.arguments else {
+        return Err("Credential spec has no arguments".to_string());
+    };
+    let Some(arg_value) = arguments.get(arg_name) else {
+        return Err(format!("Credential spec has no {}-argument", arg_name));
+    };
+    if let ArgumentValue::Int(int_value) = arg_value {
+        Ok(*int_value)
+    } else {
+        Err(format!(
+            "Credential spec has an unexpected value for {}-argument",
+            arg_name
+        ))
+    }
+}
 
-        let owner_arg = "owner";
-        let Some(value) = arguments.get(owner_arg) else {
-            return Err(format!("Credential spec has no {}-argument", owner_arg));
-        };
-        let ArgumentValue::String(owner_text) = value else {
-            return Err(format!(
-                "Credential spec has an unexpected value for {}-argument",
-                owner_arg
-            ));
-        };
-        let Ok(owner) = Principal::from_text(owner_text) else {
-            return Err(format!(
-                "Credential spec has an invalid value for {}-argument",
-                owner_arg
-            ));
-        };
-        Ok((group_name.to_string(), owner))
+fn get_string_arg_value(arg_name: &str, spec: &CredentialSpec) -> Result<String, String> {
+    let Some(arguments) = &spec.arguments else {
+        return Err("Credential spec has no arguments".to_string());
+    };
+    let Some(arg_value) = arguments.get(arg_name) else {
+        return Err(format!("Credential spec has no {}-argument", arg_name));
+    };
+    if let ArgumentValue::String(str_value) = arg_value {
+        Ok(str_value.clone())
     } else {
         Err(format!(
-            "Credential {} is not supported",
-            spec.credential_type.as_str()
+            "Credential spec has an unexpected value for {}-argument",
+            arg_name
         ))
     }
 }
 
+fn check_arg_number(expected_arg_number: usize, spec: &CredentialSpec) -> Result<(), String> {
```

Comment:
Could you rename to `check_arg_amount` or `check_arg_quantity`? I confused the objective when reading it.